modules/account-lib/src/coin/algo/utils.ts

```diff
@@ -258,6 +259,66 @@ export class Utils implements BaseUtils {
   protected mnemonicFromSeed(seed: Uint8Array) {
     return algosdk.masterDerivationKeyToMnemonic(seed);
   }
+    
+  /**
+   * Build correct fee info and fee rate for Algorand transactions.
+   *
+   // eslint-disable-next-line jsdoc/require-param-type
+   * @param tx {Object} required fields or building fee info.
+   * @param feeRate {number} required fee rate for building fee info.
+   * @returns {Object} The fee information for algorand txn.
+   */
+  getFeeData(tx, feeRate: number): any {
+    const size = this.getTransactionByteSize(tx);
+    let feeInfo = this.validateFeeInfo(size, { feeRate: feeRate, fee: undefined, baseFactor: 1 });
+    if (feeInfo.fee < ALGORAND_MINIMUM_FEE) {
+      feeInfo = this.validateFeeInfo(size, { feeRate: undefined, fee: ALGORAND_MINIMUM_FEE, baseFactor: 1 });
+      feeRate = feeInfo.feeRate;
+    }
+    return { feeInfo, feeRate };
+  }
+
+  /**
+   * Check values for feeInfo building
+   *
+   * @param size {number} required size to check.
+   * @param feeData {Object} required fees to check.
+   * @returns {Object} with fee data validated.
+   */
+  protected validateFeeInfo(size: number, { feeRate, fee, baseFactor = 1000 }): any {
+    const fee_size = this.positiveInt(size);
```

Comment:
It will be preferred to have explicit one time validation vs accessing the value everytime via a wrapper that has validation as side-effect.

My suggestion is to not have positiveInt method and have explicit one time validation, such as:
if (!Number.isSafeNumber(fee) && fee < 0) {
    // throw error
}